source/adapters/level_zero/command_buffer.cpp

```diff
@@ -135,6 +160,35 @@ ur_exp_command_buffer_handle_t_::~ur_exp_command_buffer_handle_t_() {
   for (auto &ZeFence : ZeFencesList) {
     ZE_CALL_NOCHECK(zeFenceDestroy, (ZeFence));
   }
+
+  auto ReleaseIndirectMem = [](ur_kernel_handle_t Kernel) {
+    if (IndirectAccessTrackingEnabled) {
+      // urKernelRelease is called by CleanupCompletedEvent(Event) as soon as
+      // kernel execution has finished. This is the place where we need to
+      // release memory allocations. If kernel is not in use (not submitted by
+      // some other thread) then release referenced memory allocations. As a
+      // result, memory can be deallocated and context can be removed from
+      // container in the platform. That's why we need to lock a mutex here.
+      ur_platform_handle_t Plt = Kernel->Program->Context->getPlatform();
```

Comment:
```suggestion
      ur_platform_handle_t Platform = Kernel->Program->Context->getPlatform();
```
Nit but I think this shortening is unnecessary.